File: examples/tutorial-scala/src/main/scala/tutorial_scala/controller/twitter/TweetController.scala
package tutorial_scala.controller.twitter

import org.slim3.controller.Navigation
import org.slim3.util.RequestMap
import org.slim3scala.controller.Controller

import tutorial_scala.service.TwitterService

class TweetController extends Controller {
  private val service = new TwitterService

  protected def run(): Navigation = {
    service.tweet(new RequestMap(request))
    redirect(basePath)
  }
}

File: examples/tutorial-java/src/main/java/tutorial_java/controller/twitter/TweetController.java
package tutorial_java.controller.twitter;

import org.slim3.controller.Controller;
import org.slim3.controller.Navigation;
import org.slim3.util.RequestMap;

import tutorial_java.service.TwitterService;

public class TweetController extends Controller {

    private TwitterService service = new TwitterService();

    @Override
    public Navigation run() throws Exception {
        service.tweet(new RequestMap(request));
        return redirect(basePath);
    }
}
